### DISCUSSION TITLE
How do I set a handle as invisible (and have the edge go all the way to the border of the node)

### DISCUSSION THREAD

**Discussion by gl3nn** [xyflow/xyflow#3513] [Q&A] [answered]
I haven't been able to figure out how to set a handle to 0 height and width. I went through all the code and it seems to always take into account the dimensions of the handle to calculate where the edge starts and ends should be. I haven't figured out how to set this width and height to 0. (Except maybe digging deep into the store and changing some handle bounds using symbols and stuff, but that seems overly complicated and prone to breaking on version update.)

Currently I changed the styling of the handle to be the same as the edge, which from a distance looks fine, but I find it super odd that I just can't change the size of the handles. (And then I would set it to 0).

**Comment by gl3nn** [comment]
Should this be asked somewhere else?

**Reply by bcakmakoglu** [reply, marked as answer]
No, your question was probably just overlooked 😄 

This css rule should do what you're asking for, right?

```css
.react-flow__handle {
  opacity: 0;
  height: 0;
  width: 0;
  top: auto;
  bottom: auto;
}
```

[Here's a sandbox.](https://codesandbox.io/s/friendly-architecture-8dmqyp?file=/src/styles.css:196-288)

**Reply by gl3nn** [reply]
Hmm this does work! It's just odd that I tried many variations on that. 😅 It would make the handle disappear yes, but then the edge didn't reach the node. Thanks!

**Comment by gl3nn** [comment]
Hmm actually it doesn't work! It causes the handle (thus the spot where the edge connects to the node) to always be at the top, even when I specify `Position.Bottom`.

**Reply by bcakmakoglu** [reply]
Are you sure that `top: auto` is set and is not overwritten?
If necessary you might have to force it with something like `!important`

**Reply by gl3nn** [reply]
I ended up adding (not a 100% like this of course):
```
.react-flow__handle-bottom {
    bottom: 0;
 }
```
So that the bottom handle is at the bottom. The top handle was in the right position. (I don't use any left or right handles.)

`top: auto` is applied fine but the `bottom: auto` caused the handle to always be at the top of the node. I'm satisfied with my current solution now though.

**Reply by bcakmakoglu** [reply]
Yeah I was gonna say maybe you need to be more specific for each position

```css
.react-flow__handle-bottom {
	opacity: 0;
  	height: 0;
  	width: 0;
  	top: 100%;
  	bottom: 0;
```

Depending on if you want the "invisible" handle to be inside or outside the node border you'd have to use `auto` or `100%` for `top`

**Reply by gl3nn** [reply]
Yeah I definitely wanted it to be on the inside; and I would not have figured out that `auto` would have been the solution for that. This discussion can be closed. Thanks!
